code/datums/ai/basic_mobs/basic_ai_behaviors/run_away_from_target.dm

```diff
@@ -33,17 +33,33 @@
 
 /datum/ai_behavior/run_away_from_target/proc/plot_path_away_from(datum/ai_controller/controller, atom/target)
 	var/turf/target_destination = get_turf(controller.pawn)
-	for (var/i in 1 to run_distance)
-		var/turf/test_destination = get_ranged_target_turf_direct(controller.pawn, target, range = i, offset = 180)
-		var/list/airlocks = SSmachines.get_machines_by_type_and_subtypes(/obj/machinery/door/airlock)
-		if (test_destination.is_blocked_turf(exclude_mobs = TRUE, source_atom = controller.pawn, ignore_atoms = airlocks))
+	var/static/list/offset_angles = list(45, 90, 135, 180, 225, 270)
+	for(var/angle in offset_angles)
+		var/turf/test_turf = get_furthest_turf(controller.pawn, angle, target)
+		if(isnull(test_turf))
+			continue
+		var/distance_from_target = get_dist(target, test_turf)
+		if(distance_from_target <= get_dist(target, target_destination))
+			continue
+		target_destination = test_turf
+		if(distance_from_target == run_distance) //we already got the max running distance
 			break
-		target_destination = test_destination
+
 	if (target_destination == get_turf(controller.pawn))
 		return FALSE
 	set_movement_target(controller, target_destination)
 	return TRUE
 
+/datum/ai_behavior/run_away_from_target/proc/get_furthest_turf(atom/source, angle, atom/target)
+	var/turf/return_turf
+	var/list/airlocks = SSmachines.get_machines_by_type_and_subtypes(/obj/machinery/door/airlock)
+	for(var/i in 1 to run_distance)
+		var/turf/test_destination = get_ranged_target_turf_direct(source, target, range = i, offset = angle)
+		if(test_destination.is_blocked_turf(exclude_mobs = !source.density, source_atom = source, ignore_atoms = airlocks))
+			break
```

Comment:
i'm wondering if it wouldn't be better to just `return null` here instead just so we can be 100% explicit. would you also mind dmdoccing this proc to say that it will either return a valid turf or null because that might trip people up who aren't already familiar with this proc's functionality

Reply:
> i'm wondering if it wouldn't be better to just `return null` here instead just so we can be 100% explicit. would you also mind dmdoccing this proc to say that it will either return a valid turf or null because that might trip people up who aren't already familiar with this proc's functionality

i dont want to return null here, im trying to return the furthest non-empty turf. when we meet a blocked turf, we return the empty turf before it

Reply:
oh i misread that then